lib/new_relic/agent/instrumentation/redis.rb

```diff
@@ -5,43 +5,17 @@
 require 'new_relic/agent/datastores'
 require 'new_relic/agent/datastores/redis'
 
-module NewRelic
-  module Agent
-    module Instrumentation
-      module Redis
-        extend self
-
-        UNKNOWN = "unknown".freeze
-        LOCALHOST = "localhost".freeze
-
-        def host_for(client)
-          client.path ? LOCALHOST : client.host
-        rescue => e
-          NewRelic::Agent.logger.debug "Failed to retrieve Redis host: #{e}"
-          UNKNOWN
-        end
-
-        def port_path_or_id_for(client)
-          client.path || client.port
-        rescue => e
-          NewRelic::Agent.logger.debug "Failed to retrieve Redis port_path_or_id: #{e}"
-          UNKNOWN
-        end
-      end
-    end
-  end
-end
+require_relative 'redis/instrumentation'
+require_relative 'redis/chain'
+require_relative 'redis/prepend'
 
 DependencyDetection.defer do
   # Why not :redis? newrelic-redis used that name, so avoid conflicting
   named :redis_instrumentation
-
-  depends_on do
-    defined? ::Redis::VERSION
-  end
-
+  configure_with :redis
+  
   depends_on do
-    NewRelic::Agent.config[:disable_redis] == false
+    defined?(::Redis) && defined?(::Redis::VERSION)
```

Comment:
Shouldn't we also still be checking for the config option being disabled for redis?

Reply:
It's checked before the execute block is called.  In dependency_detection.rb...

```ruby
    def allowed_by_config?
      !(disabled_configured? || deprecated_disabled_configured?)
    end

    def check_dependencies
      return false unless allowed_by_config? && dependencies

      dependencies.all? do |dep|
        begin
          dep.call
        rescue => err
          NewRelic::Agent.logger.error( "Error while detecting #{self.name}:", err )
          false
        end
      end
    end
```

Reply:
OH! I thought we were supposed to be _replacing_ things like `NewRelic::Agent.config[:disable_redis] == false` in the depends_on block with allowed_by_config?, I didn't realize the new way was hooked up to a different place.

Reply:
We're definitely supposed to reference the deprecated `:disable_foo` when we introduce the new `:instrumentation.foo` setting by setting the new attribute's default to `instrumentation_value_of(:disable_foo, ...)`. But otherwise, yes, I wired it into the Dependency Detection so that all instrumentation followed the same rules for enabling, disabling, and toggling prepend and method-chaining.